operator/internal/controller/webhook_controller.go

```
@@ -216,53 +216,61 @@ func (r *WebhookController) CheckOrUpdateWebhookCertSecret(ctx context.Context,
 		if errors.IsNotFound(err) {
-			err := r.Create(ctx, validatingWebhookConfiguration)
-			if err != nil && !errors.IsAlreadyExists(err) { // race condition, ignore
-				return false, err
-			}
-		} else {
-			return false, err
+			return false, nil
 		}
-	} else {
-		if compareValidatingWebhookConfigurations(existingValidatingWebhookConfiguration, validatingWebhookConfiguration) {
-			existingValidatingWebhookConfiguration.Webhooks = validatingWebhookConfiguration.Webhooks
-			err := r.Update(ctx, existingValidatingWebhookConfiguration)
-			if err != nil {
-				return false, err
-			}
+		return false, err
+	}
+
+	needUpdate := false
+	for i := range existingValidating.Webhooks {
+		if len(existingValidating.Webhooks[i].ClientConfig.CABundle) == 0 {
+			existingValidating.Webhooks[i].ClientConfig.CABundle = caBundle
+			needUpdate = true
+		}
+	}
+	if needUpdate {
+		if err := r.Update(ctx, existingValidating); err != nil {
+			return false, err
+		} else {
+			changed = true
 		}
 	}
 
 	// MutatingWebhookConfiguration
-	mutatingWebhookConfiguration := webhookMutatingWebhookConfiguration(r.namespace, r.opts.ServiceName, secret)
-	existingMutatingWebhookConfiguration := &admissionregistrationv1.MutatingWebhookConfiguration{}
-	err = r.Get(ctx, types.NamespacedName{Name: mutatingWebhookConfiguration.Name}, existingMutatingWebhookConfiguration)
-	if err != nil {
+	mutatingName := webhookMutatingWebhookConfiguration(r.namespace, r.opts.ServiceName, secret).GetName()
+	existingMutating := &admissionregistrationv1.MutatingWebhookConfiguration{}
+	if err := r.Get(ctx, types.NamespacedName{Name: mutatingName}, existingMutating); err != nil {
 		if errors.IsNotFound(err) {
-			err := r.Create(ctx, mutatingWebhookConfiguration)
-			if err != nil && !errors.IsAlreadyExists(err) { // race condition, ignore
-				return false, err
-			}
-		} else {
-			return false, err
+			return changed, nil
```

maybe we should return a wrapped error that says something about the creation handled by helm chart.